bin/reth/src/commands/import_receipts.rs

```diff
@@ -97,47 +92,83 @@ impl ImportReceiptsCommand {
-                file_client;
-
-            total_decoded_receipts += total_receipts_chunk;
-
+            // It is possible for the first receipt returned by the file client to be block #1
+            // rather than the genesis block. In this case, we just prepend empty receipts to the
+            // current list of receipts. When initially writing to static files, the provider
+            // expects the first block to be block zero. So, if the first block returned by the file
+            // client is block #1, we insert empty receipts for the genesis block.
+            if first_block == 1 {
+                // prepend the first empty receipts
+                receipts.insert(0, vec![]);
+                // this ensures the bundle state and static file producer start at zero
+                first_block = 0;
+                // we count this as decoded so the partial import check later does not error if this
+                // branch is executed
+                total_decoded_receipts += 1;
+            }
+
+            // We're reusing receipt writing code internal to
+            // `BundleStateWithReceipts::write_to_storage`, so we just use a default empty
+            // `BundleState`.
             let bundled_state =
                 BundleStateWithReceipts::new(Default::default(), receipts, first_block);
 
             let static_file_producer =
                 static_file_provider.get_writer(first_block, StaticFileSegment::Receipts)?;
 
+            // finally, write the receipts
             bundled_state.write_to_storage::<<DatabaseEnv as Database>::TXMut>(
                 &tx,
                 Some(static_file_producer),
                 OriginalValuesKnown::Yes,
             )?;
 
-            static_file_provider.commit()?
+            static_file_provider.commit()?;
         }
 
-        let total_imported_receipts = tx.entries::<tables::Receipts>()?;
+        if total_decoded_receipts == 0 {
+            error!(target: "reth::cli", "No receipts were imported, ensure the receipt file is valid and not empty");
+            return Ok(())
+        }
 
-        if total_imported_receipts != total_decoded_receipts {
+        // compare the highest static file block to the number of receipts we decoded
+        let total_imported_receipts = static_file_provider
+            .get_highest_static_file_block(StaticFileSegment::Receipts)
+            .expect("static files must exist after ensuring we decoded more than zero");
+
+        if total_imported_receipts != total_decoded_receipts as u64 {
             error!(target: "reth::cli",
                 total_decoded_receipts,
                 total_imported_receipts,
                 "Receipts were partially imported"
             );
         }
 
-        tx.commit()?;
-
-        info!(target: "reth::cli", "Receipt file imported");
+        info!(target: "reth::cli", total_imported_receipts, "Receipt file imported");
```

Comment:
if we import blocks first, we should make sure here that number of receipts == number of txes imo. Maybe we don't fail in case it's intentional, but at least warn the user?

Reply:
> if we import blocks first, we should make sure here that number of receipts == number of txes imo. Maybe we don't fail in case it's intentional, but at least warn the user?

makes sense

Reply:
receipts == number of txes + 6 duplicates